packages/cli/src/commands/setup/deploy/providers/aws-serverless.js

```diff
@@ -16,6 +16,7 @@ service: app
 
 plugins:
   - serverless-dotenv-plugin
```

Comment:
dot env files are going to be included by default in the next serverless version, but the functionality can be opted into now, plus helps remove a dependency.

```suggestion
useDotenv: true

plugins:
```

if this suggestion is accepted it also needs to be removed from the setup command install, see:

```javascript
// packages/cli/src/commands/setup/deploy/providers/aws-serverless.js:87
export const apiDevPackages = [
  '@netlify/zip-it-and-ship-it',
  'serverless-dotenv-plugin', // should be removed too!
]
```

Reply:
Note that in #3782 I added a [stage](https://github.com/redwoodjs/redwood/pull/3782/files#diff-f18dd32c39f000ab0ad47befa798fbf3c534c2f963f75be5ae82e0fc61a0dc30R38-R43) argument. Related to this as when deploying serverless will look for a `.env.{stage}` file before falling back to the `.env` file if non exists.

Reply:
> dot env files are going to be included by default in the next serverless version

Hi, quick note: `useDotenv: true` will work in Serverless Framework v2 and v3 (future next major) in the same way.

That means that you will need to keep `useDotenv: true` in serverless.yml if you want `.env` files to be loaded.

Note that the `serverless-dotenv-plugin` plugin works slightly differently from the built-in feature, so pick the one you prefer based on your requirements.

Reply:
Usually, secrets are indeed better stored outside of the filesystem (because it makes sharing them across the team hard & risky, and there's always the risk of committing them/publishing them somewhere).

Common approaches with AWS (and Serverless Framework) are:

- store secrets in AWS SSM Parameter Store or AWS Secrets Manager, and reference them from serverless.yml (https://www.serverless.com/framework/docs/providers/aws/guide/variables/#reference-variables-using-aws-secrets-manager)
- use Serverless Dashboard parameters: https://www.serverless.com/framework/docs/guides/parameters

Reply:
Adding either of those options into the setup tutorial is a good idea.

Reply:
What's action should be taken from this conversation @thedavidprice?

I think maybe updating https://redwoodjs.com/docs/deploy#aws-serverless-deploy to include setting up the serverless dashboard with ci/cd and secrets/env-vars? should I make an issue? (and then resolve that issue 😝 )